### ISSUE TITLE
Improperly displayed tray icon [linux]

### ISSUE THREAD

**Issue by threaz** [meetfranz/plugins#87]
I'm having issue with tray icon of Franz. I'm running Linux Manjaro with KDE. 
![franz](https://cloud.githubusercontent.com/assets/9368662/20463168/08c0f80a-af2e-11e6-97b2-0b464588c783.png)
The black square is the place where some nice looking Franz icon should be shown.

**Comment by jantman** [comment]
I'm seeing this exact issue as well intermittently on Arch Linux with KDE. When I receive a notification, the black box turns to a red box.

**Comment by jcmarchi** [comment]
Here what I found guys: if you execute Franz via command line using "sudo", the icon shows correctly.

`sudo ./Franz`

For some reason, looks like the application lacks permission to update the icon in KDE taskbar (Plasma 5.x) when running without sudo access.

So far, no side effect running it with sudo, other than have it more stable and with the tray icon nicely displayed. :)

**Comment by jantman** [comment]
Hmm, interesting. Maybe if I have a chance, I'll try ``strace``-ing Franz with both sudo and non-sudo, and see if I can figure out an obvious difference. Or just looking at my kde/plasma and xsession logs...

For me at least, I do *not* think permissions are the problem. I get the black icon like everyone else has posted, or a red icon when I have notifications, but I also sometimes see this (left-most icon here):

![spectacle x17395](https://cloud.githubusercontent.com/assets/273454/21679306/5e8969dc-d311-11e6-9055-ae739f5d5423.png)

I'm pretty sure that when I look closely, that's really just a tiny part of one of the icons. I've come to believe that the reason many people see all black, and some people see all red, is because most linux distros are expecting small (64x64?? 32x32??) tray icons, and the png files in ``resources/app.asar.unpacked/assets/tray/linux/`` are all 1024x1024 pixels. I'm not sure why running under sudo would fix this, unless it's a matter of Franz trying to detect the screen resolution in a way that a normal user can't...

Also, FWIW, regarding that fix... there's no way I'm running closed-source proprietary software as root on my system...

**Comment by threaz** [comment]
Your claim about linux distros expecting lower size images is true, @jantman . Some time ago I replaced images in `franz-bin/resources/app.asar.unpacked/assets/tray/linux` with those sitting in `light` folder and the problem is partially gone. Small icon, same as @jcmarchi mentioned earlier, now appears in my tray instead of black square. However, the ugly red numbers(same as @jantman mentioned) are still displayed when the message arrives. Is this matter of desktop environment we are using? Do you use KDE too?

**Comment by jantman** [comment]
Yeah, I use KDE as well. I'll need to look into what KDE recommends in terms of icon sizing...

**Comment by jcmarchi** [comment]
I am glad my findings triggered this discussion. Please, let me know if
there is anything else I can do to help...

I am basically running Kubuntu 16.10.

Julio Marchi ©
Technology Executive & Entrepreneur
(586) 996-3939
jcmarchi@gmail.com

This e-mail and any attachments may contain information which is
confidential, proprietary, privileged or otherwise protected by law. The
information is solely intended for the named addressee (or a person
responsible for delivering it to the addressee). If you are not the
intended recipient of this message, you are not authorized to read, print,
retain, copy or disseminate this message or any part of it. If you have
received this e-mail in error, please notify the sender immediately by
return e-mail and delete it from your computer.​

**Comment by whites11** [comment]
Any news on this? I'm having the same issue on arch with franz 4.0.4 beta and kde

**Comment by simonkrekels** [comment]
Using sudo works for the icon, but the notifications stopped showing. Sadly, the notifications were largely the reason I installed Franz in the first place.

**Comment by cribari** [comment]
Same problem here (Antergos Linux + KDE Plasma 5.9.1).

**Comment by pwoji** [comment]
Same problem on Manjaro + KDE Plasma 5.9

**Comment by paulozullu** [comment]
Yeah! I just resized the files under `Franz/resources/app.asar.unpacked/assets/tray/linux/` and `Franz/resources/app.asar.unpacked/assets/` to 32x32 to fix this.


![screenshot_20170226_153414](https://cloud.githubusercontent.com/assets/14080486/23342450/4a36fba6-fc39-11e6-8152-2e0150f5a32b.png)
![screenshot_20170226_153600](https://cloud.githubusercontent.com/assets/14080486/23342451/4a4088d8-fc39-11e6-8db6-44e51a1b8e61.png)

**Comment by jcmarchi** [comment]
Nice!

Thanks for sharing it... I've just tried and it worked for me as well

For those who are unfamiliar with how to do it via command line in Linux,
here how to:

First, exit Franz. Then run the following commands:


*$ cd
[Franz-Intallation-Location]/Franz/resources/app.asar.unpacked/assets/**$ *
*zip images-backup.zip *.png**$ *
*mogrify -resize '32x32' *.png**$ *
*cd tray/linux/**$ *
*zip images-backup.zip *.png**$ **mogrify -resize '32x32' *.png*

These commands will backup the images before converting them to the right
size.

You will be required to have ImageMagick installed. If you don have it, you
should (highly useful):

*$ **sudo apt update*
*$ **sudo apt install imagemagick*

I hope it gets fixed in the next version of Franz. Meanwhile, the above
workaround should suffice.

Julio Marchi ©
Technology Executive & Entrepreneur
(586) 996-3939
jcmarchi@gmail.com

This e-mail and any attachments may contain information which is
confidential, proprietary, privileged or otherwise protected by law. The
information is solely intended for the named addressee (or a person
responsible for delivering it to the addressee). If you are not the
intended recipient of this message, you are not authorized to read, print,
retain, copy or disseminate this message or any part of it. If you have
received this e-mail in error, please notify the sender immediately by
return e-mail and delete it from your computer.

**Comment by leandrw** [comment]
What about a definitive fix?

**Comment by AlphaBootis** [comment]
I'm seeing the same issue on KDE Neon. Definitive fix for this would be great.

**Comment by GiulioGrimani** [comment]
Same issue on Ubuntu 16.04 using Unity. Even tried to resize icons, but still no luck. Anybody having the same issue with same setup?

**Comment by glaubersm** [comment]
Tray icon is still black on KDE Plasma 5.10 running franz 4.0.4. We need a definitive fix.

**Comment by crocodile2u** [comment]
I don't have a tray icon at all. Ubuntu 17.04 with Unity. [Settings -> Leave Franz running in notification area when window is closed] is checked

**Comment by aroblu94** [comment]
Same issue as @crocodile2u once upgraded from Ubuntu 16.10 to 17.04

**Comment by crocodile2u** [comment]
Since then, I already tried Franz on Ubuntu 16.10 which came preinstalled with my new laptop - the tray icon worked perfectly. Uprgaded to 17.04 - tray icon disappeared

**Comment by aroblu94** [comment]
The workaround is to make a new entry in Startup Applications like this `env XDG_CURRENT_DESKTOP=Unity /opt/franz/Franz` and disable the in-app option `start on boot`. Unfortunately Franz window will pop up at boot, but it's not a big deal
